### PR TITLE
[Sparse] Add tests for sparse examples; Migrate from mock_sparse to mock_sparse2

### PR DESCRIPTION
**Pull request by jermainewang**
## Description

Part of #4948 .

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by mufeili** [comment]
As discussed before, change `from dgl.mock_sparse2 import ...` to `import dgl.mock_sparse2 as ...`.

File: examples/sparse/hypergraphatt.py [hunk 1]
```diff
@@ -112,5 +123,13 @@ def main():
                 refresh=False,
             )
 
-if __name__ == '__main__':
-    main()
+    print(f"Test acc: {test_acc:.3f}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Hypergraph Attention Example")
```
Comment (mufeili) [inline examples/sparse/hypergraphatt.py:130]: Why do we still use `argparse` only for this example?

File: examples/sparse/hypergraphatt.py [hunk 2]
```diff
@@ -2,13 +2,17 @@
...
+import argparse
+
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
-from torchmetrics.functional import accuracy
 import tqdm
+from torchmetrics.functional import accuracy
+
+import dgl.mock_sparse as dglsp
```
Comment (mufeili) [inline examples/sparse/hypergraphatt.py:13]: We should be able to migrate this example to `mock_sparse2`.

**Review by mufeili — APPROVED** [review]
Minor comments

File: examples/sparse/hypergraphatt.py (hunk 2 above)
Comment (jermainewang) [reply examples/sparse/hypergraphatt.py:13]: Yes, I've done that locally. Waiting for the bspmm/bsddmm PR to be merged and will push all the changes later.

File: examples/sparse/hypergraphatt.py (hunk 1 above)
Comment (jermainewang) [reply examples/sparse/hypergraphatt.py:130]: Because it's too slow for unit test. Training 500 epochs takes 10+ mins on CPU.

**Pull request merged**